### PR TITLE
sdk bug

### PR DESCRIPTION
**Pull request by tanlethanh**

### CODE REVIEW

File: apps/web/scripts/kernel/messaging.ts [hunk 1]
```diff
@@ -38,23 +39,37 @@ export const initializeMessaging = async (): Promise<void> => {
 		const callbackRegistry: Record<string, MessengerCallback> = {};
 
 		runtime.onConnect.addListener((port) => {
-			const handleInComingMessage = async (message: EncryptedMessage) => {
+			console.log('Connecting with:', port.sender?.url);
```
Comment (fluctlight-kayaba) [inline apps/web/scripts/kernel/messaging.ts:42]: remove this log

File: apps/web/scripts/kernel/messaging.ts [hunk 2]
```diff
@@ -38,23 +39,37 @@ export const initializeMessaging = async (): Promise<void> => {
...
+			const handleInComingMessage = async (
+				message: EncryptedMessage | MessagePayload,
+			) => {
+				console.log('On message');
```
Comment (fluctlight-kayaba) [inline apps/web/scripts/kernel/messaging.ts:46]: remove log

File: apps/web/scripts/kernel/messaging.ts [hunk 3]
```diff
@@ -38,23 +39,37 @@ export const initializeMessaging = async (): Promise<void> => {
...
 				if (registeredCallback) {
 					if (isEncrypted) {
 						const key = await modules.encryptionKeyVault.get(port.name);
-						const decrypted = await decryptMessage(message, key);
-
+						const decrypted = await decryptMessage(
+							message as EncryptedMessage,
+							key,
+						);
+						console.log(
```
Comment (fluctlight-kayaba) [inline apps/web/scripts/kernel/messaging.ts:57]: remove or make it display on development only

File: apps/web/scripts/kernel/messaging.ts [hunk 4]
```diff
@@ -38,23 +39,37 @@ export const initializeMessaging = async (): Promise<void> => {
...
+							`\tMessage info: \n\t - from: ${decrypted.from}, \n\t - type: ${decrypted.type} \n\t - requestId: ${decrypted.requestId}`,
+						);
 						registeredCallback?.(decrypted, port);
 					} else {
+						const payload = message as MessagePayload;
+						console.log(
```
Comment (fluctlight-kayaba) [inline apps/web/scripts/kernel/messaging.ts:63]: remove log

File: packages/core/utils/entityTypes.ts [hunk 5]
```diff
@@ -82,6 +82,6 @@ export interface Collectible {
 
 export interface TrustedDomain {
 	trusted: boolean;
-	timestamp: string;
 	connectCount: number;
+	connect: boolean;
```
Comment (fluctlight-kayaba) [inline packages/core/utils/entityTypes.ts:86]: does this mean `connected`?

Comment (tanlethanh) [reply packages/core/utils/entityTypes.ts:86]: yeah will rename it

File: apps/web/scripts/kernel/messaging.ts (hunk 1 above)
Comment (tanlethanh) [reply apps/web/scripts/kernel/messaging.ts:42]: Might we put this log here? Each time debug with sdk or service worker, I always need to know what happens. So if it has description, I want to put it here for later

Comment (fluctlight-kayaba) [reply apps/web/scripts/kernel/messaging.ts:42]: we may need an util that logs but conditional, e.g only on development - would be okay if we leave it this way for now.

**Review by fluctlight-kayaba — APPROVED** [review]

**Pull request merged**